creator/refferalTokens/models.py

```diff
@@ -0,0 +1,55 @@
+import uuid
+from django.db import models
+from creator.studies.models import Study
+from django.contrib.auth.models import Group
+
+User = get_user_model()
+
+
+class ReferralToken(models.Model):
+    """
+    An ReferralToken will be exchanged by a new or existing user in order to
+    be added to a studies with a given role.
+    """
+
+    id = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
+
+    email = models.EmailField(
+        max_length=254,
+        unique=True,
+        help_text="The email that the token link will be sent to",
+    )
+
+    claimed = models.BooleanField(
+        default=False, help_text="If the token has been used"
+    )
+
+    is_valid = models.BooleanField(
+        default=True,
+        help_text="If the token has already been claimed or if it has expired",
+    )
+
+    study = models.ForeignKey(
+        Study,
+        related_name="studys",
+        help_text="List of studies that the user will be added to",
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+
+    role = models.CharField(max_length=100, blank=False,)
+
+    groups = models.ManyToManyField(
+        Group,
+        verbose_name=_("groups"),
+        blank=True,
+        help_text="The role the user will assume in these studies",
+        related_name="groups",
```

Comment:
I think you want `referral_tokens` here as that is how you would refer to the token from the group objects.